module-2/backend/week_one.md

```diff
@@ -5,30 +5,105 @@ Fork this respository. Answer the questions to the best of your ability. Try to
 Note: When you're done, submit a PR. 
 
 1. List the five common HTTP verbs and what the purpose is of each verb.
+
+* DELETE: delete , GET: render a view, POST: submit content, PUT: update content, PATCH: sililar to PUT, used in Rails.
 2. What is Sinatra?
+
+* Sinatra is a web-application framework. It allows the developer to configure things to his liking more than Rails.
 4. What is MVC?
+
+* Model, View, Controller. Model: holds our logic; Controller: serves as our router; View: displays content
+
 5. Why do we follow conventions when creating our actions/path names in our Sinatra routes?
+
+* To be in sync with what Rails does? 
+
 6. What types of variables are accessible in our view templates without explicitly passing them?
+
+* instance
+
 7. Given the following block of code, how would I pass an instance variable `count` with a value of `1` to my `index.erb` template?
   
   ```ruby
   get '/horses' do
     erb :index
   end
+  
+  get '/names' do
+    erb :index, :locals => {:name => params[:Mr.Ed]}
+  end
+  
+  index.erb
+  <% @count %>
   ```
 
 8. In the same code block, how would I pass a local variable `name` with a value of `Mr. Ed`?
 9. What's the purpose of ERB?
+
+* Make our Ruby code readible by the browser.
+
 10. Why do I need a development AND test database?
+
+* You don't want your production database having content from your test database. In other words, you wouldn't want the samples you create in testing to end up in production.
+
 11. What's responsive design?
+
+* Mobile friendly. More specifically, responsive design is meant to keep our pages looking as similiar as possible on both mobile and desktop browsers.
+
 12. What is CRUD and why is it important?
+
+* Create, Read, Update, Delete. It's the convention Rails follows for building web-applications. 
 13. What does HTTP stand for? 
+
+* Hyper-Text-Transfer-Protocol
+
 14. What are the two ways to interpolate Ruby in an ERB view template? What's the difference between these two ways?
+
+* <%= %> The '=' tells us that the content inside is displayed.
+* <% %> The lack of the '=' tells us that the content inside will not be displayed.
+
 15. What's an ORM?
+
+* Object Relational Model
+
 16. What's the most commonly used ORM?
+
+* Actice Record
+
 17. Let's say we have an application with restaurants. There are seven verb + path combinations necessary to provide full CRUD functionality for our restaurant application. List each of the seven combinations, and explain what each is for.
+
+* get erb :index 
+
+* post erb :index 
+
+* put erb :index 
+
+* get erb :show 
+
+* delete erb :index
+
+* post erb :show 
+
+* get erb :edit 

```

Comment:
Regarding the routes: we were looking for the URL patterns, so for example a `GET` request to `/restaurants` should show us a list of all restaurants. A `POST` to `/restaurants` will create a new restaurant, etc.